server/command.go

```diff
@@ -0,0 +1,233 @@
+		return cmdError(args.ChannelId, "A link for this channel already exists, please remove unlink the channel before you link a new one")
+	}
+
+	_, err := p.msteamsAppClient.GetChannel(parameters[0], parameters[1])
+	if err != nil {
+		return cmdError(args.ChannelId, "MS Teams channel not found.")
+	}
+
+	channelsLinked[channel.TeamId+":"+channel.Id] = ChannelLink{
+		MattermostTeam:    channel.TeamId,
+		MattermostChannel: channel.Id,
+		MSTeamsTeam:       parameters[0],
+		MSTeamsChannel:    parameters[1],
+	}
+
+	channelsLinkedData, err = json.Marshal(channelsLinked)
+	if err != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	appErr = p.API.KVSet("channelsLinked", channelsLinkedData)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	p.restart()
+
+	return &model.CommandResponse{
+		ResponseType: model.CommandResponseTypeEphemeral,
+		Text:         "The msteams channel is now linked to this mattermost channel",
+		Username:     botDisplayName,
+		IconURL:      msteamsLogoURL,
+	}, nil
+}
+
+func (p *Plugin) executeUnlinkCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	channel, appErr := p.API.GetChannel(args.ChannelId)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the current channel information.")
+	}
+
+	canLinkChannel := channel.Type == model.ChannelTypeOpen && p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PermissionManagePublicChannelProperties)
+	canLinkChannel = canLinkChannel || (channel.Type == model.ChannelTypePrivate && p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PermissionManagePrivateChannelProperties))
+	if !canLinkChannel {
+		return cmdError(args.ChannelId, "Unable to unlink the channel, you has to be a channel admin to unlink it.")
+	}
+
+	channelsLinkedData, appErr := p.API.KVGet("channelsLinked")
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the linked channels information, please try again.")
+	}
+	var channelsLinked map[string]ChannelLink
+	json.Unmarshal(channelsLinkedData, &channelsLinked)
+	_, ok := channelsLinked[channel.TeamId+":"+channel.Id]
+	if !ok {
+		return cmdError(args.ChannelId, "Link doesn't exists.")
+	}
+
+	delete(channelsLinked, channel.TeamId+":"+channel.Id)
+	channelsLinkedData, err := json.Marshal(channelsLinked)
+	if err != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	appErr = p.API.KVSet("channelsLinked", channelsLinkedData)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	p.restart()
+
+	return &model.CommandResponse{
+		ResponseType: model.CommandResponseTypeEphemeral,
+		Text:         "The msteams channel is no longer linked to this mattermost channel",
+		Username:     botDisplayName,
+		IconURL:      msteamsLogoURL,
+	}, nil
```

Comment:
If we use the `SendEphemeralPost` method, and return an empty `&model.CommandResponse{}`, then we can use the MSTeams bot for the ephemeral response

Reply:
I'm trying to do that but is showing me the message sent by "system" instead of the bot. I'm not sure what is the advantage here about using the bot instead of the command response.